src/components/v2/Button/index.tsx

```diff
@@ -1,42 +1,48 @@
 /** @jsxImportSource @emotion/react */
 import React from 'react';
-import MuiButton, { ButtonProps } from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+
 import { Icon } from 'components/v2/Icon';
 import useStyles from './styles';
+import { Variant } from './types';
 
-export interface IButtonProps extends ButtonProps {
-  className?: string;
+export interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   loading?: boolean;
-  fillWidth?: boolean;
+  fullWidth?: boolean;
+  small?: boolean;
+  variant?: Variant;
 }
 
 export const Button = ({
   className,
   loading,
   disabled,
-  startIcon = null,
   fullWidth = false,
-  ...restProps
+  small = false,
+  variant = 'primary',
+  children,
+  ...otherProps
 }: IButtonProps) => {
-  const styles = useStyles({ fullWidth });
+  const styles = useStyles({ fullWidth, variant, small });
+
   return (
-    <MuiButton
-      css={styles}
+    <button
+      css={styles.button}
       className={className}
-      startIcon={loading ? <Icon name="loading" size="28px" /> : startIcon}
       disabled={loading || disabled}
-      disableRipple
-      {...restProps}
-    />
+      type="button"
```

Comment:
It is probably that we will want to set this to "submit" for certain use cases

Reply:
Yeah in this case it can be overwritten through the props we pass down. I originally wanted to simply assign a default value to `type`, but TypeScript complained it has to assigned as a string on the tag itself.